File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
using namespace std;

bool check(int n) {
    if (n < 2) return false;
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int solution(vector<int> nums) {
    int answer = 0;
    for (int i = 0; i < nums.size(); i++) {
        for (int j = i + 1; j < nums.size(); j++) {
            for (int n = j + 1; n < nums.size(); n++) {
                if (check(nums[i] + nums[j] + nums[n]))answer++;
            }
        }
    }
    return answer;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.kt
class Solution {
    fun isPrime(n: Int):Boolean {
        if (n < 2) return false
        var i = 2
        while (i * i <= n) {
            if (n % i++ == 0) return false
        }
        return true
    }

    fun solution(nums: IntArray): Int {
        var answer = 0
        for(i in nums.indices)
            for(j in i+1 until nums.size)
                for(n in j+1 until nums.size)
                    if(isPrime(nums[i]+nums[j]+nums[n]))answer++
        return answer
    }
}